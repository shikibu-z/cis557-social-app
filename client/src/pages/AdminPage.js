/* eslint-disable import/no-extraneous-dependencies */
import React, { useEffect, useState } from 'react';
import { withRouter } from 'react-router';
import NavBar from '../components/NavBar';
import '../stylesheets/AdminPage.css';
import AdminBox from '../components/AdminBox';
import AdminNotification from '../components/AdminNotification';

import { getGroupInfo, getAdminNotes } from '../api/fetchers';
import GroupAnalyticBox from '../components/GroupAnalyticBox';

function AdminPage() {
  const id = localStorage.getItem('currUser');
  const [group, setGroup] = useState();
  const [adminNotes, setAdminNotes] = useState([]);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(async () => {
    const url = window.location.href;
    const groupid = (url.split('/').slice(-2)[0]);
    const response = await getGroupInfo(id, groupid);
    setGroup(response.data);
    const res = await getAdminNotes(groupid);
    setAdminNotes(res.data);
    setIsLoading(false);
  }, []);

  return (
    !isLoading
    && (
    <div>
      <NavBar isLogin={0} />
      <div className="adminHeader">
        <img src={group.group.profilePhoto} alt="group-img" />
        <div id="adminNameWrapper">
          {group.group.groupName}
        </div>
      </div>
      <hr />
      <h2>Group Introduction</h2>
      <div className="adminGroupIntro">
        {group.group.groupInfo}
      </div>
      <h2>Analytics</h2>
      <GroupAnalyticBox groupId={group.group.idGroups} />
      <h2>Notifications</h2>
      {adminNotes.map((n) => (
        <AdminNotification note={n} group={group} />
      ))}
      <AdminBox group={group.group.idGroups} />
    </div>
    )
  );
}

export default withRouter(AdminPage);
