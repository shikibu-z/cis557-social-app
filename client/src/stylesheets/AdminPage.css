.adminHeader {
    display: flex;
    flex-direction: row;
    align-items: center;
    margin-left: 10%;
}
#adminNameWrapper {
    display: block;
    font-family: Roboto;
    font-size: 32px;
    font-style: normal;
    font-weight: bold;
    margin-left: 20%;
}

.adminHeader img{
    display: block;
    width: 150px;
    height: 150px;
    border-radius: 50%;
    overflow: hidden;
    margin: 1%;
}

h2 {
    margin-left: 10%;
}

.adminGroupIntro {
    margin-left: 10%;
    border: 1px solid black;
    width: 50%;
    height: 100px;
    margin-bottom: 2%;
}

