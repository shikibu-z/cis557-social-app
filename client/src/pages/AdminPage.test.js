/*eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import AdminPage from './AdminPage';

test('Chat page matches snapshot', () => {
  const component = renderer.create(<BrowserRouter><AdminPage /></BrowserRouter>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
