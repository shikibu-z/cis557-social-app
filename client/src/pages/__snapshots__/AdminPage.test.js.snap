// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat page matches snapshot 1`] = `null`;
